File: src/main/cs/00/Problem0048.cs
using System;
using System.Diagnostics;
using System.Numerics;
using Euler.test.cs;

namespace Euler.main.cs.de.swerik.euler._00
{
    public class Problem0048
    {
        public Problem0048()
        {
            var stopWatch = new Stopwatch();
            stopWatch.Start();
            BigInteger result = 0;

            // Solution:
            for (var i = 1; i < 1000; i++) result += BigInteger.Pow(i, i);


            stopWatch.Stop();
            var elapsedTime = stopWatch.Elapsed.ToString();
            Console.WriteLine("Result:\t" + result.ToString().Substring(result.ToString().Length - 10) + "\tTime:\t" +
                              (double.Parse(elapsedTime.Substring(elapsedTime.LastIndexOf(":") + 1, 2)) >= 1
                                   ? double.Parse(elapsedTime.Substring(elapsedTime.LastIndexOf(":") + 1)) + " s"
                                   : double.Parse(elapsedTime.Substring(elapsedTime.IndexOf(".") + 1)) / 10_000 +
                                     " ms"));
            if (ProblemTest.DoBenchmark)
                Benchmark.AddTime(48, double.Parse(elapsedTime.Substring(elapsedTime.IndexOf(".") + 1)) / 10_000);
        }
    }
}

File: src/main/java/de/swerik/euler/_00/Problem0048.java
package de.swerik.euler._00;

import java.math.BigInteger;

public class Problem0048 {
    public Problem0048() {
        long startTime = System.nanoTime();
        BigInteger result = BigInteger.ZERO;

        // Solution:
        for (int i = 1; i < 1000; i++) result = result.add(new BigInteger(i + "").pow(i));


        long timeToResolve = System.nanoTime() - startTime;
        System.out.println("Result:\t" + result.toString().substring(result.toString().length() - 10) + "\tTime:\t" + (((double) timeToResolve / 1_000_000) > 1000 ?
                (((double) timeToResolve / 1_000_000_000) + "s") :
                (((double) timeToResolve / 1_000_000) + "ms")));
    }
}
